File: Adapter/python/AdapterPattern.py
class Target:
    def print_weak(self):
        raise NotImplementedError

    def print_strong(self):
        raise NotImplementedError


class Adaptee:
    def __init__(self, message):
        self.message = message

    def print_paren(self):
        print("(" + self.message + ")")

    def print_asterisk(self):
        print("*" + self.message + "*")


class Adapter(Target):
    def __init__(self, adaptee):
        self.adaptee = adaptee

    def print_weak(self):
        self.adaptee.print_paren()

    def print_strong(self):
        self.adaptee.print_asterisk()


def run():
    target = Adapter(Adaptee("Adaptee is adpated to Target by Adapter"))
    target.print_weak()
    target.print_strong()

run()

File: Adapter/ecma2015/AdapterPattern.js
'use strict';
class Target {
    printWeak() {
        throw Error("printWeak method should be overridden");
    }

    printStrong() {
        throw Error("printStrong method should be overridden");
    }
}

class Adapter extends Target {
    constructor(adaptee) {
        super();
        this.adaptee = adaptee;
    }

    printWeak() {
        this.adaptee.printParen();
    }

    printStrong() {
        this.adaptee.printAsterisk();
    }
}

class Adaptee {
    constructor(message) {
        this.message = message;
    }

    printParen() {
        console.log(`(${this.message})`);
    }

    printAsterisk() {
        console.log(`*${this.message}*`);
    }
}

(()=> {
    "use strict";
    let target = new Adapter(new Adaptee("Adaptee is adapted to Target by Adapter"));
    target.printWeak();
    target.printStrong();
})();